crates/sui-types/src/crypto.rs

```diff
@@ -33,21 +35,129 @@ use crate::error::{SuiError, SuiResult};
 use crate::sui_serde::{Base64, Readable, SuiBitmap};
 pub use enum_dispatch::enum_dispatch;
 
-// Comment the one you want to use
-
 // Authority Objects
 pub type AuthorityKeyPair = Ed25519KeyPair;
 pub type AuthorityPrivateKey = Ed25519PrivateKey;
 pub type AuthorityPublicKey = Ed25519PublicKey;
 pub type AuthoritySignature = Ed25519Signature;
 pub type AggregateAuthoritySignature = Ed25519AggregateSignature;
 
-// Account Objects
+// TODO(joyqvq): prefix these types with Default, DefaultAccountKeyPair etc
 pub type AccountKeyPair = Ed25519KeyPair;
 pub type AccountPublicKey = Ed25519PublicKey;
 pub type AccountPrivateKey = Ed25519PrivateKey;
 pub type AccountSignature = Ed25519Signature;
 
+#[enum_dispatch]
```

Comment:
You're not using any of the features of `#[enum_dispatch]`, here.
You can go with manual definitions (as you do here), or actually use the type (which would require you to define the enum where the traits are defined).